src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[3];
+  const separatorTo = toFormat[3];
+
+  const dateParts = date.split(separatorFrom);
+  const dateObject = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateObject[part] = dateParts[index];
+  });
+
+  if (dateObject['YYYY']) {
+    dateObject['YY'] = dateObject['YYYY'].slice(-2);
+  } else if (dateObject['YY']) {
+    const year = parseInt(dateObject['YY'], 10);
+
+    dateObject['YYYY'] =
+      year < 30 ? '20' + dateObject['YY'] : '19' + dateObject['YY'];
+  }
+
+  const newDateParts = toFormat.slice(0, 3).map((part) => dateObject[part]);
```

Comment:
Instead of slicing the 'toFormat' array, you can directly map over the first three elements. This would avoid creating an unnecessary shallow copy of the array slice.